components/automate-gateway/gateway/middleware/authz/authz.go

```diff
@@ -106,21 +96,37 @@ func (c *client) IsAuthorized(ctx context.Context, subjects []string, resource,
 	})
 	if err != nil {
 		if status.Convert(err).Code() == codes.FailedPrecondition {
-			return nil, err
+			return nil, false, err
 		}
 		log.WithError(err).Error("error authorizing request")
-		return nil, status.Errorf(codes.PermissionDenied,
-			"error authorizing action %q on resource %q for members %q: %s",
+		// something went wrong in some unexpected way -- we'll "fail shut", i.e., deny authorization
+		return nil, false, errors.Errorf("error authorizing action %q on resource %q for members %q: %s",
 			action, resource, subjects, err.Error())
 	}
 	projects := filteredResp.Projects
 
-	return &resp{
-		ctx:        auth_context.NewContext(ctx, subjects, projects, resource, action),
-		authorized: len(projects) != 0,
-	}, nil
+	return auth_context.NewContext(ctx, subjects, projects, resource, action),
+		len(projects) != 0,
+		nil
 }
 
+// FilterAuthorizedPairs drives authz introspection
+// Q: There's no projects here! This can't be right!
+// A(msorens): The short answer to that is introspection is not yet project-aware.
```

Comment:
@msorens hope you don't mind that I've liberated your useful comment from slack; it'll be easier to find here for future explorers.